azavea/raster-foundry
# Add datasource creation modal

## Overview

There's currently no way to add a datasource from the UI; this adds one.

### Checklist


### Demo

![createdatasource1](https://user-images.githubusercontent.com/447977/27090419-268a9b54-502b-11e7-9b0a-1985a5c879f5.png)
![createdatasource2](https://user-images.githubusercontent.com/447977/27090426-29a4767a-502b-11e7-8cb3-0a93e262287c.png)
![createdatasource3](https://user-images.githubusercontent.com/447977/27090433-2cc33be8-502b-11e7-832c-b6f55585219e.png)

### Notes

It would be nice to add color composite editing into the modal but that would require refactoring some existing code and also might be too complex for many users. Instead, I set it to create all datasources with a natural-color composite as a default, since that should satisfy the needs of many users. We should revisit this once we have user feedback.

It would also be nice to integrate this into the project-creation flow; I've left the `modalInstance` and `resolve` bindings in place with that in mind even though they're currently unused.

## Testing Instructions

 * Visit `/imports`; make sure that you see a "Create new datasource" button in the right sidebar.
 * Click the button and make sure that creating a new datasource works.
 * Also make sure that you get an error if there's no name.
 * Make sure that the links in the Success! pane work (subject to #1980)
 * Make sure that the newly-created datasource has a natural-color composite

Closes #1690
